tests/fuzz/target_fuzz.go

```diff
@@ -0,0 +1,13 @@
+// +build gofuzz
+
+package fuzz
+
+// FuzzUnmarshal tests unmarshaling
+func FuzzUnmarshal(fuzz []byte) int {
+	data := &Data{}
+	err := data.UnmarshalJSON(fuzz)
+	if err != nil {
+		return 0
+	}
```

Comment:
I would also Marshal the result back to `[]byte` here to kinda test that as well.